Index: imms-3.1.0-rc6/configure.ac
===================================================================
--- imms-3.1.0-rc6.orig/configure.ac	2008-10-06 20:20:43.000000000 -0600
+++ imms-3.1.0-rc6/configure.ac	2008-10-07 07:47:03.000000000 -0600
@@ -101,6 +101,7 @@ if test "$with_pcre" = "no"; then
     AC_MSG_ERROR([PCRE required and missing.])
 else
     LIBS="$LIBS $pcre_LIBS"
+    CPPFLAGS="$CPPFLAGS $pcre_CFLAGS"
 fi
 
 if test "$enable_analyzer" != "no"; then
@@ -139,18 +140,19 @@ if test "$enable_analyzer" != "no"; then
 fi
 
 AC_CHECK_TOOL(OBJCOPY, objcopy)
-if test "x$OBJCOPY" = "x"; then
-    AC_MSG_ERROR("objcopy from GNU binutils >= 2.11.90 not found")
-fi
 AC_CHECK_TOOL(OBJDUMP, objdump)
-dnl extract target and architecture if objdump was found
-if test "x$OBJDUMP" = "x"; then :; else
+if test "x$OBJCOPY" = "x"; then
+    AC_MSG_WARN("objcopy from GNU binutils >= 2.11.90 not found")
+    AC_DEFINE(BROKEN_OBJCOPY,, [broken objcopy])
+else
+    dnl extract target and architecture if objdump was found
     AC_MSG_CHECKING(for object target)
     octarget=`$OBJDUMP --info | sed -ne '2p'` # extract first target
     AC_MSG_RESULT($octarget)
     AC_MSG_CHECKING(for object architecture)
     ocarch=`$OBJDUMP --info | sed -ne '4p'` # extract corresponding arch
     AC_MSG_RESULT($ocarch)
+    AC_SUBST(SVM_SIMILARITY, svm-similarity-data.o)
 fi
 AC_SUBST(OBJCOPYTARGET, $octarget)
 AC_SUBST(OBJCOPYARCH, $ocarch)
Index: imms-3.1.0-rc6/immsd/immsd.cc
===================================================================
--- imms-3.1.0-rc6.orig/immsd/immsd.cc	2008-10-06 20:20:43.000000000 -0600
+++ imms-3.1.0-rc6/immsd/immsd.cc	2008-10-06 20:20:45.000000000 -0600
@@ -231,6 +231,7 @@ void quit(int signum)
     if (loop)
         g_main_quit(loop);
     loop = 0;
+    LOG(INFO) << "goodbye." << endl;
     signal(signum, SIG_DFL);
 }
 
Index: imms-3.1.0-rc6/Makefile
===================================================================
--- imms-3.1.0-rc6.orig/Makefile	2008-10-06 20:20:43.000000000 -0600
+++ imms-3.1.0-rc6/Makefile	2008-10-06 20:20:45.000000000 -0600
@@ -9,6 +9,7 @@ configure: configure.ac
 	autoheader
 	aclocal
 	autoconf
+	[ ! -x config.status ] || ./config.status --recheck
 
 immsconf.h: configure
 	$(error Please run the "configure" script)
@@ -31,6 +32,8 @@ dist: immsconf.h distclean
 	rm -rf /tmp/imms-$(VERSION)/
 	mv configure autogen.sh
 
+all: configure
+
 vars.mk:;
 
 %:
Index: imms-3.1.0-rc6/rules.mk
===================================================================
--- imms-3.1.0-rc6.orig/rules.mk	2008-10-06 20:20:43.000000000 -0600
+++ imms-3.1.0-rc6/rules.mk	2008-10-06 20:20:45.000000000 -0600
@@ -14,9 +14,8 @@ link = $(CXX) $(filter-out %.a,$1) $(fil
 %.o: %.c; $(call compile, $(CC), $<, $@, $($*-CFLAGS) $(CFLAGS) $($*-CPPFLAGS) $(CPPFLAGS))
 %: %.o; $(call link, $^ $($*-OBJ) $(LIBS), $@, $($*-LIBS) $(LDFLAGS))
 %.so:
-	$(CXX) $^ $($*-OBJ) $($*-LIBS) $(LIBS) \
-	    $(LDFLAGS) \
-            -shared -Wl,-z,defs,-soname,$@ -o $@
+	$(CXX) -flat_namespace -undefined suppress -o $@ -bundle $^ $($*-OBJ) $($*-LIBS) $(LIBS) \
+	    $(LDFLAGS) -o $@
 
 %-data.o: %
 	$(OBJCOPY) -I binary -O $(OBJCOPYTARGET) -B $(OBJCOPYARCH) --rename-section .data=.rodata,alloc,load,readonly,data,contents $< $@
Index: imms-3.1.0-rc6/model/model.cc
===================================================================
--- imms-3.1.0-rc6.orig/model/model.cc	2008-10-06 20:20:43.000000000 -0600
+++ imms-3.1.0-rc6/model/model.cc	2008-10-07 07:41:25.000000000 -0600
@@ -81,15 +81,20 @@ public:
         string filename = get_imms_root("svm-similarity");
         if (file_exists(filename))
         {
-            LOG(INFO) << "Overriding the built in model with " << filename;
+            LOG(INFO) << "Overriding the built in model with " << filename << endl;
             model.reset(new DiskXFile(filename.c_str(), "r"));
         }
         else
         {
+#ifndef BROKEN_OBJCOPY
             static const size_t data_size = &_binary____data_svm_similarity_end
                 - &_binary____data_svm_similarity_start;
             model.reset(new MemoryXFile(
                         &_binary____data_svm_similarity_start, data_size));
+#else
+            LOG(ERROR) << "Couldn't find the built-in model at " << filename << endl;
+            exit(1);
+#endif
         }
         normalizer.load(model.get());
         svm.loadXFile(model.get());
Index: imms-3.1.0-rc6/Torch3.diff
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ imms-3.1.0-rc6/Torch3.diff	2008-10-06 20:20:45.000000000 -0600
@@ -0,0 +1,22 @@
+---
+ makefile |   13 +++++++++++++
+ 1 file changed, 13 insertions(+)
+
+Index: Torch3/makefile
+===================================================================
+--- /dev/null	1970-01-01 00:00:00.000000000 +0000
++++ Torch3/makefile	2008-10-06 20:13:28.000000000 -0600
+@@ -0,0 +1,13 @@
++# frontend makefile for IMMS users
++PREFIX=/usr/local
++
++libs/Darwin_opt_float/libtorch.a: Darwin.cfg
++	./xmake
++
++Darwin.cfg: config/Darwin.cfg
++	sed -e 's/packages =/packages = gradients distributions kernels/' $< > $@
++
++install: libs/Darwin_opt_float/libtorch.a
++	mkdir -p $(PREFIX)/{lib,include/torch}
++	cp {gradients,distributions,kernels}/*.h $(PREFIX)/include
++	cp $< $(PREFIX)/lib
Index: imms-3.1.0-rc6/build/Makefile
===================================================================
--- imms-3.1.0-rc6.orig/build/Makefile	2008-10-06 18:29:42.000000000 -0600
+++ imms-3.1.0-rc6/build/Makefile	2008-10-07 07:45:46.000000000 -0600
@@ -15,7 +15,7 @@ training: training_data train_model
 libimmscore.a: $(call objects,../immscore)
 	$(AR) $(ARFLAGS) $@ $(filter %.o,$^)
 
-libmodel.a: $(call objects,../model) svm-similarity-data.o
+libmodel.a: $(call objects,../model) $(SVM_SIMILARITY)
 	$(AR) $(ARFLAGS) $@ $(filter %.o,$^)
 
 immstool: immstool.o libmodel.a libimmscore.a mfcckeeper.o
Index: imms-3.1.0-rc6/vars.mk.in
===================================================================
--- imms-3.1.0-rc6.orig/vars.mk.in	2008-10-06 19:52:06.000000000 -0600
+++ imms-3.1.0-rc6/vars.mk.in	2008-10-07 07:46:05.000000000 -0600
@@ -8,6 +8,7 @@ OBJCOPY = @OBJCOPY@
 OBJDUMP = @OBJDUMP@
 OBJCOPYTARGET = @OBJCOPYTARGET@
 OBJCOPYARCH = @OBJCOPYARCH@
+SVM_SIMILARITY= @SVM_SIMILARITY@
 exec_prefix = @exec_prefix@
 bindir = @bindir@
 datadir = @datadir@
Index: imms-3.1.0-rc6/README.osx
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ imms-3.1.0-rc6/README.osx	2008-10-07 07:49:45.000000000 -0600
@@ -0,0 +1,30 @@
+You will need the following dependencies from MacPorts (http://macports.org):
+
+    pcre
+    glib2
+    taglib
+
+    autoconf
+    automake
+    pkgconfig
+    binutils
+
+In addition, you will need Torch (http://www.torch.ch/). To install Torch,
+download and extract the source, apply Torch3.diff, then:
+
+    make
+    sudo make install
+
+Torch is now installed.
+
+Finally, in the IMMS directory just type:
+
+    ./configure
+    make
+    sudo make install
+    mkdir -p ~/.imms
+    cp data/svm-similarity ~/.imms
+
+Let me know if this doesn't work for you.
+
+-- Hans Fugal <hans@fugal.net>
